ttnn/cpp/ttnn/operations/moreh/moreh_nll_loss/moreh_nll_loss_step1/device/moreh_nll_loss_step1_device_operation.hpp

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/operations/moreh/moreh_nll_loss/moreh_nll_loss_helper.hpp"
+
+namespace ttnn::operations::moreh::moreh_nll_loss_step1 {
+
+struct MorehNllLossStep1DeviceOperation {
+    struct operation_attributes_t {
+        const std::string reduction = NONE;
+        const uint32_t ignore_index = std::numeric_limits<uint32_t>::max();
+        const DataType output_dtype;
+        const uint32_t channel_size = 1;
+        const MemoryConfig memory_config;
+        const DeviceComputeKernelConfig compute_kernel_config;
```

Comment:
I previously thought that the reason for using optional<DeviceComputeKernelConfig> in operation_attributes_t was that the build would fail without it. Is the current code building successfully?
If the build is working fine, I think it would be better to use DeviceComputeKernelConfig as it is now. I will check why optional<DeviceComputeKernelConfig> is being used in the TT side.

Reply:
I made that change because the build was successful, and all test cases passed.

Reply:
The current code is successfully built.

Reply:
I'm asking about the reason for using `optional<DeviceComputeKernelConfig>` in TT. I will share the response once I receive it.

Reply:
TT mentioned that optional depends on the use-case, so it seems it doesn't need to be optional.